integration_tests/test_domestic_migration.py

```diff
@@ -123,3 +123,88 @@ def test_only_living_people_move(simulants_on_adjacent_timesteps):
         movers = before["address_id"] != after["address_id"]
         assert after[movers]["tracked"].all()
         assert (after[movers]["alive"] == "alive").all()
+
+
+# add employer address tests
+# 2. all simulants with same employer id have same employer details (static)
+# 3. all employer ids have unique address details (static)
+
+
+def test_addresses_during_moves(simulants_on_adjacent_timesteps):
+    """Check that address details change after a move. Note that
+    this test does not distinguish between the different types of moves (eg
+    household vs individual moves)
+    """
+    unit_id_col = "employer_id"
+    address_id_col = "business_details.employer_address_id"
+    other_details_cols = ["business_details.employer_name"]
+    _test_addresses_during_moves(
+        pops=simulants_on_adjacent_timesteps,
+        unit_id_col=unit_id_col,
+        address_id_col=address_id_col,
+        other_details_cols=other_details_cols,
+    )
+
+
+def _test_addresses_during_moves(pops, unit_id_col, address_id_col, other_details_cols):
+    details_cols = [unit_id_col, address_id_col] + other_details_cols
+    for before, after in pops:
+        mask_movers = before[address_id_col] != after[address_id_col]
+        # address details change if address_id changes
+        # NOTE: we cannot assert that all details are different because some
+        # units (eg employer "not implemented") have the same name.
+        assert (
+            not (
```

Comment:
Isn't this just checking that address details have changed for at least on of the movers? We want to check that the address details have changed for all of them, so we want the `.all()` calls to be `.any()`

Reply:
Ok, I totally re-worked this test